File: src/main/java/hu/elte/alkfejl/familytodolist/app/model/Task.java
package hu.elte.alkfejl.familytodolist.app.model;

import java.sql.Timestamp;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "TASKS")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Task extends BaseEntity{
    @Column(nullable = false, unique = true)
    private String name;
    
    @Column(nullable = false, unique = true)
    private String description;
    
    @Column(nullable = false, unique = false)
    private int userId;
    
    @Column(nullable = false, unique = false)
    private int categoryId;
    
    @Column(nullable = false, unique = false)
    private int timeintervalId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, unique = false)
    private Status status;
    
    public enum Status{
        INPROGRESS, DONE
    }
    
    public void setStatus(Status status){
        this.status = status;
    }
}

File: familytodolist/src/app/models/task.ts
import { BaseModel } from './base-model';
import { User } from './user';
import { Category } from './category'
import { TimeInterval } from './time-interval'
 
export class Task extends BaseModel {

    public name: string;
    public description: string;
    public user: User;
    public category: Category;
    public timeInterval: TimeInterval;
    public status: Status;

    constructor(object) {
        super(object.id);
        Object.assign(this, object);
    }
}

export enum Status{
    INPROGRESS, DONE    
}